### PR TITLE
feat: add repository for invite service

### PR DESCRIPTION
**Pull request by skelet0no**
1. Added repository files
2. Change docstrings
3. Added methods in invite.py
4. Added db connecting python file
5. Added value not found file in errors
6. Added prisma schema
7. Added class decorator singleton
8. Small fixes in other file

### CODE REVIEW

File: invite_service/prisma/schema.prisma [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  provider = "prisma-client-py"
+}
+
+model Invite {
+  id String @id @default(uuid())
+  event_id String
+  author_id String
+  invitee_id String
+  status InviteStatus
+  deleted_at DateTime?
```
Comment (Nhsdkk) [inline invite_service/prisma/schema.prisma:16]: add created_at field

File: invite_service/pyproject.toml [hunk 2]
```diff
@@ -9,6 +9,7 @@ readme = "README.md"
 python = "^3.11"
 grpcio = "^1.62.1"
 grpcio-tools = "^1.62.1"
+invite = "^0.0.1"
```
Comment (Nhsdkk) [inline invite_service/pyproject.toml:12]: unknown package

File: invite_service/repository/invite_repository_impl.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    """
+    Class for manipulating with invite data
+
+    Attributes
+    ----------
+    _db_client : prisma.Client
+        Postgres db client.
+
+    Methods
+    -------
```
Comment (Nhsdkk) [inline invite_service/repository/invite_repository_impl.py:26]: Add get_invites_by_invitee_id method

File: invite_service/repository/invite_repository_impl.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    async get_invites(author_id)
+        Returns invites that has matches with given author id.
+    async create_invite(invite)
+        Creates new invite inside db or throws an exception.
```
Comment (Nhsdkk) [inline invite_service/repository/invite_repository_impl.py:30]: Delete throws an exception part

File: invite_service/repository/invite_repository_impl.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        ------
+        prisma.errors.PrismaError
+            Catch all for every exception raised by Prisma Client Python.
+        ValueNotFoundError
+            No invites was found for given author id.
+
+        """
+        db_invites: Optional[
+            List[PrismaInvite]
+        ] = await self._db_client.db.invite.find_many(where={"id": author_id})
```
Comment (Nhsdkk) [inline invite_service/repository/invite_repository_impl.py:67]: check that deleted_at is None

File: invite_service/repository/invite_repository_impl.py [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+            invite object.
+
+        Raises
+        ------
+        prisma.errors.PrismaError
+            Catch all for every exception raised by Prisma Client Python.
+
+        """
+        await self._db_client.db.invite.update(
+            where={"id": invite.id}, data=invite.to_dict()
```
Comment (Nhsdkk) [inline invite_service/repository/invite_repository_impl.py:107]: check that deleted_at is none

File: invite_service/repository/mock_invite_repository.py [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+        """
+        Creates invite with matching data or throws an exception
+
+        Parameters
+        ----------
+        invite : Invite
+            Invite data
+
+        """
+        invite.id = uuid4()
```
Comment (Nhsdkk) [inline invite_service/repository/mock_invite_repository.py:75]: use str(uuid4())

File: invite_service/repository/mock_invite_repository.py [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+        Updates invite with matching id or throws an exception
+
+        Parameters
+        ----------
+        invite : Invite
+            Invite data
+
+        Raises
+        ------
+        ValueError
```
Comment (Nhsdkk) [inline invite_service/repository/mock_invite_repository.py:89]: Male it raise Value Not Found Error

File: invite_service/repository/mock_invite_repository.py [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+        Deletes invite with matching id or throws an exception
+
+        Parameters
+        ----------
+        invite_id : str
+            Invite's id
+
+        Raises
+        ------
+        ValueError
```
Comment (Nhsdkk) [inline invite_service/repository/mock_invite_repository.py:106]: Make it raise value not found error

File: invite_service/src/models/invite.py [hunk 10]
```diff
@@ -54,3 +57,49 @@ def to_grpc_invite(self) -> GrpcInvite:
...
+        exclude : Optional[List[str]]
+            List of fields to exclude.
+
+        Returns
+        -------
+        Dict[str, Any]
+            Invent data dictionary.
+
+        """
+        exclude_set = set(exclude if exclude is not None else ["id"])
```
Comment (Nhsdkk) [inline invite_service/src/models/invite.py:76]: use set([]) l {"id"} in case of None

**Review by Nhsdkk — CHANGES_REQUESTED** [review]
Check review and make changes

File: invite_service/repository/mock_invite_repository.py [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+    ----------
+    _invites: List[Invite]
+        List of invites
+
+    Methods
+    -------
+    async get_invites(author_id)
+        Returns invites that has matches with given author id.
+    async create_invite(invite)
+        Creates new invite inside db or throws an exception.
```
Comment (ZixMai) [inline invite_service/repository/mock_invite_repository.py:25]: remove "or throws an exception"

File: invite_service/repository/mock_invite_repository.py [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+    async update_invite(incite)
```
Comment (ZixMai) [inline invite_service/repository/mock_invite_repository.py:26]: invite, not incite

File: invite_service/repository/mock_invite_repository.py [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+        invite : Invite
+            Invite data
+
+        Raises
+        ------
+        ValueError
+            Can't update invite with provided data
+
+        """
+        self._invites[self._invites.index(invite)] = invite
```
Comment (ZixMai) [inline invite_service/repository/mock_invite_repository.py:93]: @Nhsdkk ! **wrong way** to find invite index, fixed in my branch.
 _when you updating event in this code you are trying to find index of **NEW** event object that cant be in_ self._invites by definition.  
try to find index of old invite object that matches by id with new instead. you can use "index = next(i for i in range(len(self._invites)) if self._invites[i].id == invite.id)" and add try except block with exception StopIteration for wrong requests and raise ValueNotFoundError in the end

File: invite_service/repository/mock_invite_repository.py [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+        invite_id : str
+            Invite's id
+
+        Raises
+        ------
+        ValueError
+            Can't delete invite with provided data
+
+        """
+        index = [i for i in range(len(self._invites)) if self._invites[i].id == invite_id]
```
Comment (ZixMai) [inline invite_service/repository/mock_invite_repository.py:110]: use next() instead of list()[0], check update_invite method for the instructions

**Review by ZixMai — CHANGES_REQUESTED** [review]
Check "your own" code for mistakes pls (or just try to read it if you arent) and rename branch by adding feat/ at the beggining...

**Pull request closed without merging**